metricbeat/module/kubernetes/state_node/state_node_test.go

```diff
@@ -39,3 +61,88 @@ func TestEventMapping(t *testing.T) {
 		},
 	)
 }
+
+// testMetricSetEventsFetcher goes over the given TestCases and ensures that source Prometheus metrics gets converted
+// into the expected events when passed by the given metricset.
+// If -update_expected flag is passed, the expected JSON file will be updated with the result
+func testMetricSetEventsFetcher(t *testing.T, module, metricset string, cases TestCases) {
+	for _, test := range cases {
+		t.Logf("Testing %s file\n", test.MetricsFile)
+
+		file, err := os.Open(test.MetricsFile)
+		assert.NoError(t, err, "cannot open test file "+test.MetricsFile)
+
+		body, err := ioutil.ReadAll(file)
+		assert.NoError(t, err, "cannot read test file "+test.MetricsFile)
+
+		server := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(200)
+			w.Header().Set("Content-Type", "text/plain; charset=ISO-8859-1")
+			w.Write([]byte(body))
+		}))
+
+		server.Start()
+		defer server.Close()
+
+		config := map[string]interface{}{
+			"module":     module,
+			"metricsets": []string{metricset},
+			"hosts":      []string{server.URL},
+		}
+
+		f := mbtest.NewReportingMetricSetV2(t, config)
+		events, errs := mbtest.ReportingFetchV2(f)
+		if len(errs) > 0 {
+			t.Fatalf("Expected 0 error, had %d. %v\n", len(errs), errs)
+		}
+		assert.NotEmpty(t, events)
+
+		if *expectedFlag {
+			sort.SliceStable(events, func(i, j int) bool {
+				h1, _ := hashstructure.Hash(events[i], nil)
```

Comment:
I'm wondering why you need to sort them / why events were in different orders sometimes?

Reply:
This was copy pasted from `ptest` package but I agree that we maybe have some tricky logic here... maybe even solving specific things of different packages.